src/devices/Bmxx80/Bmx280Base.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//Ported from https://github.com/adafruit/Adafruit_BMP280_Library/blob/master/Adafruit_BMP280.cpp
+//Formulas and code examples can also be found in the datasheet http://www.adafruit.com/datasheets/BST-BMP280-DS001-11.pdf
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Threading.Tasks;
+using Iot.Device.Bmxx80.CalibrationData;
+using Iot.Device.Bmxx80.PowerMode;
+using Iot.Device.Bmxx80.Register;
+using Iot.Units;
+
+namespace Iot.Device.Bmxx80
+{
+    /// <summary>
+    /// Represents the core functionality of the Bmx280 family.
+    /// </summary>
+    public abstract class Bmx280Base : Bmxx80Base
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Bmx280Base"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        protected Bmx280Base(byte deviceId, I2cDevice i2cDevice)
+            : base(deviceId, i2cDevice)
+        {
+            _calibrationData = new Bmx280CalibrationData();
+            _calibrationData.ReadFromDevice(this);
+            _controlRegister = (byte)Bmx280Register.CONTROL;
+        }
+
+        /// <summary>
+        /// Read the temperature.
+        /// </summary>
+        /// <returns>Calculated temperature.</returns>
+        public async Task<Temperature> ReadTemperatureAsync()
+        {
+            if (ReadPowerMode() == Bmx280PowerMode.Forced)
+            {
+                await Task.Delay(GetMeasurementTimeForForcedMode(ReadTemperatureSampling()));
+            }
+
+            //Read the MSB, LSB and bits 7:4 (XLSB) of the temperature from the BMP280 registers
+            byte msb = Read8BitsFromRegister((byte)Bmx280Register.TEMPDATA_MSB);
+            byte lsb = Read8BitsFromRegister((byte)Bmx280Register.TEMPDATA_LSB);
+            byte xlsb = Read8BitsFromRegister((byte)Bmx280Register.TEMPDATA_XLSB); // bits 7:4
+
+            //Combine the values into a 32-bit integer
+            int t = (msb << 12) + (lsb << 4) + (xlsb >> 4);
+
+            return CompensateTemperature(t);
+        }
+
+        /// <summary>
+        /// Recommended wait timings from the datasheet.
+        /// </summary>
+        /// <param name="sampleMode">The <see cref="Sampling"/> to get for.</param>
+        /// <returns>The time it takes for the chip to read data in milliseconds rounded up.</returns>
+        private int GetMeasurementTimeForForcedMode(Sampling sampleMode)
+        {
+            if (sampleMode == Sampling.UltraLowPower)
```

Comment:
NIT: probably we can use the new switch c# 8 feature. Like:
```c#
return sampleMode switch
{
  Sampling.UltraLowPower => 7,
  Sampling.LowPower => 9,
  Sampling.Standard => 14
  //..
  _ => 0
};
```

Reply:
Oh that's cool!